src/helpers/xml-builder.js

```diff
@@ -1430,116 +1479,339 @@ const buildTableCellProperties = (attributes) => {
-        vNode.properties.style.border
-      );
-
-      attributes.tableCellBorder = {
-        top: borderSize,
-        left: borderSize,
-        bottom: borderSize,
-        right: borderSize,
-        colors: {
-          ...setUpDirectionalBorderColor(borderColor),
-        },
-        strokes: {
-          ...setUpDirectionalBorderStroke(borderStroke),
-        },
-      };
-    }
-  }
+const fixupTableCellBorder = (vNode, attributes, tableBorderOptions = {}, rowIndexEquivalent) => {
+  let { color } = tableBorderOptions;
+  const tableCellStyles = vNode.properties.style || {};
   if (!attributes.tableCellBorder) {
     attributes.tableCellBorder = { strokes: {}, colors: {} };
   }
-  if (vNode.properties.style['border-top'] && vNode.properties.style['border-top'] === '0') {
-    attributes.tableCellBorder = {
-      ...attributes.tableCellBorder,
-      top: 0,
-    };
-  } else if (vNode.properties.style['border-top'] && vNode.properties.style['border-top'] !== '0') {
-    // eslint-disable-next-line no-use-before-define
-    const [borderSize, borderStroke, borderColor] = cssBorderParser(
-      vNode.properties.style['border-top']
-    );
-    attributes.tableCellBorder = {
-      ...attributes.tableCellBorder,
-      top: borderSize,
-      colors: { ...attributes.tableCellBorder.colors, top: borderColor },
-      strokes: { ...attributes.tableCellBorder.strokes, top: borderStroke },
-    };
-  }
-  if (vNode.properties.style['border-left'] && vNode.properties.style['border-left'] === '0') {
-    attributes.tableCellBorder = {
-      ...attributes.tableCellBorder,
-      left: 0,
-    };
-  } else if (
-    vNode.properties.style['border-left'] &&
-    vNode.properties.style['border-left'] !== '0'
-  ) {
-    // eslint-disable-next-line no-use-before-define
-    const [borderSize, borderStroke, borderColor] = cssBorderParser(
-      vNode.properties.style['border-left']
-    );
-    attributes.tableCellBorder = {
-      ...attributes.tableCellBorder,
-      left: borderSize,
-      colors: { ...attributes.tableCellBorder.colors, left: borderColor },
-      strokes: { ...attributes.tableCellBorder.strokes, left: borderStroke },
-    };
+  const tableCellStyleKeys = Object.keys(tableCellStyles);
+
+  const rowIndexEquivalentFirst = rowIndexEquivalent.indexOf('first')
+  const rowIndexEquivalentLast = rowIndexEquivalent.indexOf('last')
+  // here this means that 
+  if (rowIndexEquivalentFirst !== -1) {
+    const indexOfBorder = tableCellStyleKeys.lastIndexOf('border')
+    const indexOfBorderTop = tableCellStyleKeys.lastIndexOf('border-top')
+    const indexOfBorderTopWidth = tableCellStyleKeys.lastIndexOf('border-top-width')
+    const indexOfBorderTopStyle = tableCellStyleKeys.lastIndexOf('border-top-style')
+    const indexOfBorderTopColor = tableCellStyleKeys.lastIndexOf('border-top-color')
+    let indexes = []
+
+    if (indexOfBorderTopStyle !== -1 && tableCellStyles['border-top-style'] === 'hidden') {
```

Comment:
Can you explain hidden vs none?

Reply:
Hidden is a special case. If given, no borders are shown no matter the priority. Thats why separately written